assets/js/base/hooks/use-store-cart-item.js

```diff
@@ -22,32 +24,60 @@ import { useStoreCart } from './use-store-cart';
  */
 export const useStoreCartItem = ( cartItemKey ) => {
 	const { cartItems, cartIsLoading } = useStoreCart();
-	const cartItem = cartItems.filter( ( item ) => item.key === cartItemKey );
-
-	const results = useSelect(
-		( select, { dispatch } ) => {
+	const [ cartItem, setCartItem ] = useState( {
+		key: '',
+		isLoading: true,
+		cartData: {},
+		quantity: 0,
```

Comment:
This 0 is a special value that we use for forcing an update during load. We should use a const instead of literal, and we could even use a more distinct placeholder value (e.g. -1).